src/lib/storage/constraints/concatenated_constraint_checker.hpp

```diff
@@ -51,15 +51,13 @@ class ConcatenatedConstraintChecker : public RowTemplatedConstraintChecker<Tuple
     return rows;
   }
 
-  virtual bool preprocess_chunk(std::shared_ptr<const Chunk> chunk) {
+  virtual void prepare_read_chunk(std::shared_ptr<const Chunk> chunk) {
```

Comment:
I still don't fully get this. You are storing the segments to avoid the atomic load, correct? I don't have the graphs handy right now - is this something that has a significant benefit? If yes, this should be documented.

Reply:
That, and so we can just iterate through these segments (without checking anything else) to fetch the values for each row.